parsec/serde/schema.py

```diff
@@ -99,17 +100,27 @@ def get_obj_type(self, obj):
 
     def _get_schema(self, type):
         if self._instantiated_schemas is None:
+            # Assert all schemas key are Enum:
+            if not all([isinstance(k, Enum) for k in self.type_schemas]):
+                raise ValueError("type_schemas key can only be an Enum")
             self._instantiated_schemas = {
                 k: v if isinstance(v, Schema) else v() for k, v in self.type_schemas.items()
             }
+
             if self.fallback_type_schema and not isinstance(self.fallback_type_schema, Schema):
                 self._instantiated_fallback_schema = self.fallback_type_schema()
 
+        if not isinstance(type, Enum):
+            for k in self._instantiated_schemas:
```

Comment:
would be simpler to do this when building `_instantiated_schemas`:
```python
self._instantiated_schemas: Dict[Union[Enum, str], Schema]  # Important to type this attribute given it key pattern is exotic
for k, v in self.type_schemas.item():
    schema_instance = v if isinstance(v, Schema) else v()
    self._instantiated_schemas[k] = schema_instance
    self._instantiated_schemas[k.value] = schema_instance
```

Reply:
fixed